bindings/wasm/src/account/wasm_account.rs

```diff
@@ -0,0 +1,365 @@
+
+    let promise: Promise = future_to_promise(async move {
+      account
+        .as_ref()
+        .borrow()
+        .resolve_identity()
+        .await
+        .map(WasmResolvedDocument::from)
+        .map(Into::into)
+        .wasm_result()
+    });
+    promise.unchecked_into::<PromiseResolvedDocument>()
+  }
+
+  /// Removes the identity from the local storage entirely.
+  ///
+  /// Note: This will remove all associated document updates and key material - recovery is NOT POSSIBLE!
+  #[wasm_bindgen(js_name = deleteIdentity)]
+  pub fn delete_identity(self) -> Promise {
+    let account = self.0;
+    let did = account.as_ref().borrow().did().to_owned();
+    let storage = account.as_ref().borrow().storage_arc();
+    std::mem::drop(account);
+
+    let promise: Promise = future_to_promise(async move {
+      let account = AccountBuilder::new()
+        .storage(AccountStorage::Custom(storage))
+        .load_identity(did)
+        .await
+        .wasm_result();
+
+      match account {
+        Ok(a) => a.delete_identity().await.wasm_result().map(|_| JsValue::undefined()),
+        Err(e) => Err(e),
+      }
+    });
+    promise
+  }
+
+  /// Push all unpublished changes to the tangle in a single message.
+  #[wasm_bindgen]
+  pub fn publish(&mut self) -> Promise {
+    let account = self.0.clone();
+    future_to_promise(async move {
+      account
+        .as_ref()
+        .borrow_mut()
+        .publish()
+        .await
+        .map(|_| JsValue::undefined())
+        .wasm_result()
+    })
+  }
+
+  /// Push all unpublished changes to the Tangle in a single message, optionally choosing
+  /// the signing key used or forcing an integration chain update.
+  ///
+  /// @see {@link PublishOptions}
+  #[wasm_bindgen (js_name = publishWithOptions)]
+  pub fn publish_with_options(&mut self, publish_options: WasmPublishOptions) -> Promise {
+    let mut options: PublishOptions = PublishOptions::new();
+
+    if let Some(force_integration) = publish_options.forceIntegrationUpdate() {
+      options = options.force_integration_update(force_integration);
+    }
+
+    if let Some(sign_with) = publish_options.signWith() {
+      let s: String = sign_with;
+      options = options.sign_with(s);
+    }
+
+    let account = self.0.clone();
+    future_to_promise(async move {
+      account
+        .as_ref()
+        .borrow_mut()
+        .publish_with_options(options)
+        .await
+        .map(|_| JsValue::undefined())
+        .wasm_result()
+    })
+  }
+
+  /// Signs a {@link Credential} with the key specified by `fragment`.
+  #[wasm_bindgen(js_name = createSignedCredential)]
+  pub fn create_signed_credential(
+    &self,
+    fragment: String,
+    credential: &WasmCredential,
+    signature_options: WasmSignatureOptions,
+  ) -> PromiseCredential {
+    let account = self.0.clone();
+    let mut cred: Credential = credential.0.clone();
+    let options: SignatureOptions = SignatureOptions::from(signature_options);
```

Comment:
Opinion: I think it might be better to take a reference to `WasmSignatureOptions` and clone it to avoid nulling-out its pointer in Javascript.

E.g.
```suggestion
    signature_options: &WasmSignatureOptions,
  ) -> PromiseCredential {
    let account = self.0.clone();
    let mut cred: Credential = credential.0.clone();
    let options: SignatureOptions = signature_options.0.clone();
```